DHFPROD-7131: Handle entities with updated props when building join menu

wooldridge opened the pull request:
### Description

Updated properties in entities that have not been saved yet will now show up in the Join Property menu in Modeling.

To test:

1. Create a new property for an entity (e.g. Customer). Do not save the entity.
2. Create a new entity relationship that points from one entity (Order) to the entity with the new property (Customer).
3. Open the Join Property menu to confirm that the new property is included in the list of properties. (Previously, it was not.)

#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

srinathgit reviewed (approved):
Verified that the property shows up in 'Order'

srinathgit reviewed (changes requested):
PR has to be updated based on the discussion we just had

wooldridge commented:
In discussion with the team, we decided that when a new property hasn't been saved to the backend yet in Modeling, that property shouldn't be allowed as a Join Property value. The property will be displayed in the menu but be disabled (just like multiple and structured properties are disabled). 

The tooltip associated with the Join Property field currently reads:

"Structured type properties and arrays cannot be used as join properties."

I suggest updating the tooltip to the following:

"Structured type properties, arrays, and unsaved properties cannot be used as join properties."

@sbayatpur does that work for you?

wooldridge commented:
> PR has to be updated based on the discussion we just had

@srinathgit I updated the branch to show but also disable unsaved properties in the Join Property menu. Also updated tooltip to explain unsaved properties cannot be used as join properties.

srinathgit reviewed (dismissed):
Looks good !

briantang reviewed (dismissed):
Nice!
